crates/smtp/Cargo.toml

```diff
@@ -56,8 +56,11 @@ decancer = "1.6.1"
 unicode-security = "0.1.0"
 infer = "0.15.0"
 
+[dev-dependencies]
+mail-auth = { git = "https://github.com/stalwartlabs/mail-auth", features = ["test"] }  # required for mock resolver
+
 [features]
-test_mode = []
+test_mode = ["mail-auth/test"]  # required for mock resolver
```

Comment:
This avoids a pesky compile issue I've faced a few times where the test feature doesn't get loaded on `cfg(test)` builds, and should correctly only include it for `cfg(any(test, feature = "test_mode"))`. I can make a separate PR for this if desired.